lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -733,6 +740,22 @@ static Value collapseAndMoveBatchDims(Location loc, Value values, int64_t batch,
   return b.create<AtenViewOp>(loc, valuesTy, values, outDimsList);
 }
 
+// Broadcast the `values` tensor to the slice size created by the list of index
+// tensors.
+static Value broadcastValuesToSliceSize(Location loc, Value values,
+                                        llvm::ArrayRef<Value> indices,
+                                        OpBuilder b) {
+  auto valuesType = cast<ValueTensorType>(values.getType());
+  auto [broadcastShape, broadcastStaticShape] =
+      getBroadcastShape(loc, indices, b);
+  auto resultType = b.getType<Torch::ValueTensorType>(
+      broadcastStaticShape, valuesType.getOptionalDtype());
+  Value broadcastShapeList = b.create<PrimListConstructOp>(
+      loc, Torch::ListType::get(b.getType<Torch::IntType>()), broadcastShape);
+  return b.create<AtenBroadcastToOp>(loc, resultType, values,
```

Comment:
Note: in numpy's advanced indexing semantics, the shape of the slice depends on whether the index tensors are along consecutive dimensions or not. However, the `IndexPutHackedTwinOp` [does not allow optional indexing tensors](https://github.com/llvm/torch-mlir/issues/3489), so the slice shape is always the common broadcasted shape of all the index tensors.

Reply:
makes sense